### PR TITLE
Add support for SSH through Session Manager

### PR DESCRIPTION
**Pull request by awiddersheim**
It is possible to SSH through an AWS Session Manager session[1]. The
best way to achieve this is to connect using the AWS instance ID as the
destination with a `ProxyCommand` configured for SSH.

[1]: https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-getting-started-enable-ssh-connections.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -401,12 +401,13 @@ If you don't set this it will default to whatever DHCP address EC2 hands out.
...
 #### `interface`
 
-The place from which to derive the hostname for communicating with the instance.  May be `dns`, `public`, `private` or `private_dns`.  If this is unset, the driver will derive the hostname by failing back in the following order:
+The place from which to derive the hostname for communicating with the instance.  May be `dns`, `public`, `private`, `private_dns` or `id`.  If this is unset, the driver will derive the hostname by failing back in the following order:
 
 1. DNS Name
 2. Public IP Address
 3. Private IP Address
 4. Private DNS Name
+5. Instance ID (Useful for [SSH through Session Manger][ssh_over_ssm])
```
Comment (multani) [inline README.md]:
```suggestion
5. Instance ID (Useful for [SSH through Session Manager][ssh_over_ssm])
```

Comment (awiddersheim) [reply README.md]: Thanks.

**Comment by jaronsummers** [comment]
Hello! Any idea when this might be merged?

**Comment by kchristensen** [comment]
+1 on this, we're about to disable our bastion servers and the lack of Session Manager support is going to make using kitchen-ec2 very tricky.

**Comment by tas50** [comment]
Thanks @awiddersheim

**Comment by e-moshaya** [comment]
I created a feature request for the same change in Hashicorp Packer. I'm hoping this will be supported too https://github.com/hashicorp/packer/issues/8242

**Pull request merged**